lib/elixir_internal_certification/guardian.ex

```diff
@@ -0,0 +1,22 @@
+defmodule ElixirInternalCertification.Guardian do
+  use Guardian, otp_app: :elixir_internal_certification
+
+  alias ElixirInternalCertification.Account.Accounts
+  alias ElixirInternalCertification.Account.Schemas.User
+
+  def subject_for_token(%User{id: user_id} = _resource, _options), do: {:ok, to_string(user_id)}
+  def subject_for_token(_resource, _options), do: {:error, :unhandled_resource_type}
+
+  def resource_from_claims(%{"sub" => sub} = _claims) do
+    user_id = String.to_integer(sub)
+
+    try do
+      user = Accounts.get_user_by_id!(user_id)
+      {:ok, user}
+    rescue
+      Ecto.NoResultsError -> {:error, :invalid_claims}
+    end
+  end
```

Comment:
```suggestion
  def resource_from_claims(%{"sub" => user_id} = _claims) do
    try do
      user = Accounts.get_user_by_id!(user_id)
      {:ok, user}
    rescue
      Ecto.NoResultsError -> {:error, :invalid_claims}
    end
  end
```
Don't think we need to convert it to integer do we? I think at minimum change `sub` to `id` or `user_id`

You have used try/rescue because it's an `exceptional` occurance right? The user id not existing from claims 👌

Reply:
- Alright let's keep the converting of the string

- Regarding Try/Rescue I was just wonderibg if it was better than using the non-bang version for getting a user and having this:

```elixir
case Accounts.get_user_by_id(user_id) do
    %User{} = user ->
      {:ok, user}

    nil ->
      {:error, :invalid_claims}
end
```

But yeah no worries we can keep existing